.github/workflows/sycl_windows_build_and_test.yml

```
@@ -80,9 +95,61 @@ jobs:
         cmake --build build --target check-xptifw
     - name: Install
       shell: bash
-      run: cmake --build build --target deploy-sycl-toolchain
+      run: |
+        # TODO replace utility installation with a single CMake target
+        cmake --build build --target utils/count/install
+        cmake --build build --target utils/not/install
+        cmake --build build --target utils/lit/install
+        cmake --build build --target utils/llvm-lit/install
+        cmake --build build --target install-llvm-size
+        cmake --build build --target install-llvm-cov
+        cmake --build build --target install-llvm-profdata
+        cmake --build build --target utils/FileCheck/install
+        cmake --build build --target install-compiler-rt
+        cmake --build build --target deploy-sycl-toolchain
+        # Copy FileCheck to install/bin for use by E2E tests.
+        cp build/bin/FileCheck.exe install/bin/.
+    - name: Pack toolchain
+      shell: bash
+      run: |
+        tar -czf llvm_sycl.tar.gz -C install .
     - name: Upload toolchain
       uses: actions/upload-artifact@v3
       with:
         name: sycl_windows_default
-        path: install/**/*
+        path: llvm_sycl.tar.gz
+
+  e2e-tests:
+    needs: build
+    if: ${{ inputs.lts_matrix != '[]' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJSON(inputs.lts_matrix) }}
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.runs-on }}
+    env: ${{ matrix.env }}
+    steps:
+    - uses: ilammy/msvc-dev-cmd@7315a94840631165970262a99c72cfb48a65d25d
+      with:
+        arch: amd64
+    - name: Set env
+      run: |
+        git config --system core.longpaths true
+        git config --global core.autocrlf false
+        echo "C:\Program Files\Git\usr\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+    # TODO: use cached_checkout
+    - uses: actions/checkout@v3
+      with:
+        persist-credentials: false
+        ref: ${{ inputs.build_ref || github.ref }}
```

```suggestion
        ref: ${{ inputs.build_ref || github.sha }}
```

Linux task was fixed in https://github.com/intel/llvm/pull/8913 .